docs: OFO doc updates (#278)
open-feature/open-feature-operator: docs/getting_started.md

# Getting Started

Once you have [installed the operator](./installation.md) you can follow this guide to deploy an example application demonstrating the operator.

### Deploy a `FeatureFlagConfiguration`

> This `FeatureFlagConfiguration` will be watched by the injected `flagd` container to create a twin of its flag definitions internally. If multiple configurations are supplied to `flagd` these states will be combined.

Reviewer: ```suggestion
This `FeatureFlagConfiguration` will be watched by the injected `flagd` container and used to construct its internal flag definitions state. If multiple configurations are supplied to `flagd` these states will be merged.
```

> This `FeatureFlagConfiguration` will be watched by the injected `flagd` container and used to construct its internal flag definitions state. If multiple configurations are supplied to `flagd` these states will be merged.

Reviewer 2: ```suggestion
This `FeatureFlagConfiguration` is watched by the injected `flagd` container and used to construct its internal flag definitions state. If multiple configurations are supplied to `flagd` these states will be merged.
```

I generall recommend using present tense,  imperative mood and passive voice for documentation.

So avoid "will be" and just say "is". This sentence is already passive voice, so that's fine, but avoid the tense shifts.

```yaml
apiVersion: core.openfeature.dev/v1alpha2
kind: FeatureFlagConfiguration
metadata:
  name: featureflagconfiguration-sample
spec:
  featureFlagSpec:
    flags:
      foo:
        state: "ENABLED"
        variants:
          "bar": "BAR"
          "baz": "BAZ"
        defaultVariant: "bar",
        targeting: {}
```

### Reference the deployed FeatureFlagConfiguration within a Deployment spec annotation.

> In this example we will be creating a `Deployment` containing an `busybox-curl` container, in the example below the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).

Reviewer: ```suggestion
In this example, we will be creating a `Deployment` containing an `busybox-curl` container. In the example below, the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).
```

> In this example, we will be creating a `Deployment` containing an `busybox-curl` container. In the example below, the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).

Reviewer 2: ```suggestion
In this example, a`Deployment` containing a `busybox-curl` container is created. In the example below, the `metadata.annotations` object contains the required annotations for the operator to correctly configure and inject the `flagd` sidecar into each deployed `Pod`. The documentation for these annotations can be found [here](./annotations.md).
```

Converted to passive voice (de-emphasize the subject: "we").

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: busybox-curl
  annotations:
    openfeature.dev/enabled: "true"
    openfeature.dev/featureflagconfiguration: "default/featureflagconfiguration-sample"
spec:
  replicas: 1
  selector:
    matchLabels:
      app: my-busybox-curl-app
  template:
      metadata:
        labels:
          app: my-busybox-cur-app
      spec:
        containers:
        - name: busybox
          image: yauritux/busybox-curl:latest
          ports:
          - containerPort: 80
          args:
            - sleep
            - "30000"
```

### Confirm that operator has injected the `flagd` sidecar

> Once the `deployment.yaml` has been applied, our `Pod` should be created and contain 2 containers.

Reviewer: ```suggestion
Once the `deployment.yaml` has been applied, our `Pod` should be created grouping 2 containers.
```

```
kubectl get pods -n default
```
Should give a similar output to the following
```
NAME                                                READY   STATUS              RESTARTS   AGE
busybox-curl-7bd5767999-spf7v                              0/2     ContainerCreating   0          2s
```

> If we describe our `Pod` we should should see that the injected sidecar has the following configuration:

Reviewer 2: ```suggestion
When the `Pod` is described, the injected sidecar has the following configuration:
```
